pkg/apis/k0s/v1beta1/api.go

```diff
@@ -35,6 +35,9 @@ type APISpec struct {
 	// Local address on which to bind an API
 	Address string `json:"address"`
```

Comment:
Now I feel really stupid for only realizing it now, but this parameter actually sounds like what we want to do here.

We use this parameter as the advertise address, not as the bind address. Would the following assumption hold?

"The advertise address will work when connecting to the API server locally, as well as from other peer controllers."

If yes, then the implementation could be much easier: Just fix all the places where k0s uses a hard coded localhost or 127.0.0.1 to connect to the API server by using the `Address`. This is pretty much all the places where `BindAddress` is used now. Then set `spec.api` to sth. like `{"address": "x.x.x.x", "extraArgs": {"bind-address": "x.x.x.x"}}`. And we're done. (At least that's what I'd guess :see_no_evil:)

Reply:
I noticed address myself last time I worked on this, totally forgot to add a comment about that :D I'll look into it.

Reply:
@twz123 the latest commit replaces BindAddress with Address as suggested.

Both versions (with dedicated BindAddress and with Address used for bind address) seem to work.

@gakio12 in the original PR #1038 actually asked the question:

> Should a new bindAddress option be created and change documentation for the existing address option, or should the existing address option's behavior be changed to match the existing documentation?

To which @jnummelin answered, that he prefers the dedicated option.

How would we justify (and document) the existence of both Address BindAddress? I see one possible reason to separate them: `0.0.0.0` as a bind address ([which also came up in the previous PR.](https://github.com/k0sproject/k0s/pull/1038#pullrequestreview-817786306)). If address is something I want to use to connect to the API server, then it should never be `0.0.0.0`. If address is something the server binds to, then I'd expect to be able to set `0.0.0.0`.

Reply:
The broken tests seem to be broken by the switch over to Address also being the bind address and I think that makes sense. I just attempted to change it so Address takes to role of BindAddress and merge its previous role with ExternalAddress, but that falls apart quickly.

My current understanding is:

* Address: Used internally by the cluster
* ExternalAddress: Used by clients external to the cluster
* BindAddress: The address the api server binds to

Clearly from this, if BindAddress is set, than Address must be compatible with its value, because we can't bind the API server to e.g. 10.0.0.1 and then advertise it's address as 10.0.0.2. But we can bind it to 0.0.0.0 and then advertise it as any address.

New proposal for the options:

* Address: IP used internally by the cluster and used for the `--bind-address` option in API server
* ExternalAddress: Address used by clients external to the cluster (does this one actually need to be an IP address?)
* OnlyBindToAddress: Boolean option (default false for compatibility) to allow binding the API server to only the IP address given by Address instead of 0.0.0.0/::

Address and ExternalAddress would also be validated to not be 0.0.0.0 or ::.